src/processTargets/PipelineStages.types.ts

```diff
@@ -0,0 +1,10 @@
+import { Target } from "../typings/target.types";
+import { ProcessedTargetsContext } from "../typings/Types";
+
+export interface MarkStage {
+  run(context: ProcessedTargetsContext): Target[];
+}
+
+export interface ModifierStage {
+  run(context: ProcessedTargetsContext, target?: Target): Target | Target[];
```

Comment:
how come modifiers can return a single target but marks cannot?  seems inconsistent.  I'd lean towards always requiring a list, but at the very least I'd argue for consistency

Reply:
The ones that doesn't require a list doesn't user list. I aim for simplicity.

Reply:
But I guess consistency is good as well so I'm updating